### DISCUSSION TITLE
Skonto variables in invoice document

### DISCUSSION THREAD

**Discussion by iusgit** [kimai/kimai#3012] [General]
Hi,

just wanted to ask to be sure, I haven't seen any Skonto related functionality and variables for invoicing?

If that's the case, I would like to implement that e.g. skonto-field on project level and making that field available in invoice templates in a similar manner as VAT variables.

Thanks

**Comment by kevinpapst** [comment]
Can you try to make it a bundle?
You should be able to provide decorated services for all the existing instances of `App\Invoice\CalculatorInterface`.
In the services.yaml something like 

```yaml
    KimaiPlugin\SkontoBundle\Invoice\DefaultCalculator:
        decorates: App\Invoice\Calculator\DefaultCalculator
```

And then 
```php
namespace KimaiPlugin\SkontoBundle\Invoice;

use App\Invoice\Calculator\DefaultCalculator as BaseDefaultCalculator;

class DefaultCalculator extends AbstractCalculator implements CalculatorInterface
{
    private $calculatur;

    public function __construct(BaseDefaultCalculator $calculatur)
    {
        $this->calculatur = $calculatur;
    }
    // ... your Skonto magic here ...
}
```
